### PR TITLE
ConstantQ refactor

### PR DESCRIPTION
**Pull request by dbogdanov**
Re-written erroneous implementation for the FFT-based ConstantQ algorithm and related SpectrumCQ and Chromagram algorithms.

Not tested yet, only merge after writing proper unit tests.

### CODE REVIEW

File: src/algorithms/standard/constantq.cpp [hunk 1]
```diff
@@ -42,23 +42,25 @@ const char* ConstantQ::description = DOC("This algorithm computes Constant Q Tra
...
 void ConstantQ::compute() {
 
-  const vector<complex<Real> >& fft = _fft.get();
+  const vector<Real> & frame = _frame.get();
   vector<complex<Real> >& constantQ = _constantQ.get();
 
-  if (fft.size() != _inputFFTSize) {
-    throw EssentiaException("ConstantQ: input FFT size must be equal to: ", _inputFFTSize);
+  if (frame.size() != _windowSize) {
+    throw EssentiaException("ConstantQ: input FFT size must be equal to: ", _windowSize);
```
Comment (dbogdanov) [inline src/algorithms/standard/constantq.cpp]: --> "input frame size..."

File: src/algorithms/standard/constantq.cpp [hunk 2]
```diff
@@ -70,76 +72,87 @@ void ConstantQ::configure() {
...
+  _fft->output("fft").set(_fftData);
 
   // Work only with a non-negative part of FFT as an input
-  _inputFFTSize = _windowSize/2+1;
+  _inputFFTSize = _windowSize / 2 + 1;
+
+  _sparseKernel = {};
 
-  struct SparseKernel &sk = _sparseKernel;
-  sk.i.reserve(_windowSize);
```
Comment (dbogdanov) [inline src/algorithms/standard/constantq.cpp]: What's the rationalle  not using `reserve()`? They are here to make interative ``push_back()`` calls faster.

**Comment by dbogdanov** [comment]
```Chromagram``` algorithm should be adapted accordingly. It is missing the ``window`` parameter.

File: src/algorithms/standard/constantq.cpp (hunk 2 above)
Comment (pabloEntropia) [reply src/algorithms/standard/constantq.cpp]: I just though that the code will be more clear without these references and forgot to keep the `reserves` when removing them. I'll put them back.

**Pull request merged**